api/src/services/setup/upgrade.js

```diff
@@ -0,0 +1,51 @@
+const indexerProgressService = require('./indexer-progress');
+const upgradeLogService = require('./upgrade-log');
+const installer = require('./install');
+const logger = require('../../logger');
+
+const upgrade = async (version, username, stageOnly) => {
+  if (!version) {
+    throw new Error('Version is invalid');
+  }
+
+  try {
+    await installer.prep(version, username, stageOnly);
+    safeInstall(version, stageOnly);
+  } catch (err) {
+    await upgradeLogService.setErrored();
+    throw err;
+  }
+};
+
+const safeInstall = async (version, stageOnly) => {
+  try {
+    await installer.stage(version);
+    await installer.indexStagedViews();
+    if (stageOnly) {
+      return;
+    }
+    await complete();
+  } catch (err) {
+    await upgradeLogService.setErrored();
+    logger.error('Error thrown when indexing views %o', err);
+  }
+};
+
+const complete = async () => {
```

Comment:
Can we change this to `finalize`, and maybe update the associated state to `finalizing`? `complete` can be interpreted either as it _is_ complete already, or a command to complete it.

Reply:
yes!